### PR TITLE
Share card

### PR DESCRIPTION
**Pull request by theutku**

### CODE REVIEW

File: src/components/SegmentedTabs/SegmentedTabs.tsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ export const SegmentedTabs = <T extends PropertyKey>({
     onChange?.(selectedItem?.value, selectedIndex)
   }
 
-  const handleSegmentedSwitcherPress = (tabName: T, index: number) => {
+  const handleSegmentedSwitcherPress = (_: T, index: number) => {
```
Comment (theutku) [inline src/components/SegmentedTabs/SegmentedTabs.tsx:95]: no need to change the naming for this

File: src/components/SegmentedTabs/SegmentedTabsSwitcherItem.tsx [hunk 2]
```diff
@@ -50,10 +50,12 @@ export const SegmentedTabsSwitcherItem = <T,>({
       onPress={() => onPress(item.value, index)}
       style={[styles.item, { backgroundColor: colors.primary }]}
     >
-      <Animated.View style={animatedMaskStyles} />
-      <Typography color={'white'} variant='bodyLarge'>
-        {t(item.title)}
-      </Typography>
+      <>
```
Comment (theutku) [inline src/components/SegmentedTabs/SegmentedTabsSwitcherItem.tsx:53]: Why do we need a fragment here?

File: src/components/ShareButton/ShareButton.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      const shareOptions = {
+        title: item?.pages[0].title,
+        message: item?.pages[0].title,
+        urls
+      }
+
+      try {
+        await Share.open(shareOptions)
+      } catch (error) {
+        //
```
Comment (theutku) [inline src/components/ShareButton/ShareButton.tsx:29]: you can add console.debug or console.warn here

File: src/components/elements/DatePicker/DatePicker.tsx [hunk 4]
```diff
@@ -1,14 +1,28 @@
...
+  showDatePicker: boolean
+  setShowDatePicker: (showDatePicker: boolean) => void
 }
 
-export const DatePicker: React.FC<DatePickerProps> = ({ title, value, ...rest }) => (
+export const DatePicker: React.FC<DatePickerProps> = ({ title, value, showDatePicker, setShowDatePicker, ...rest }) => (
   <View>
     {!!title && <Typography variant='bodySemiBold'>{title}</Typography>}
-    <DateTimePicker mode='date' value={value} {...rest} />
+    {Platform.OS === 'ios' ? (
```
Comment (theutku) [inline src/components/elements/DatePicker/DatePicker.tsx:14]: you should use isIos util

File: package.json [hunk 5]
```diff
@@ -53,9 +53,10 @@
     "react-native-reanimated": "^3.15.0",
     "react-native-safe-area-context": "4.10.5",
     "react-native-screens": "3.31.1",
+    "react-native-share": "^11.0.3",
     "react-native-svg": "^15.5.0",
     "react-native-svg-transformer": "^1.5.0",
-    "react-native-webview": "^13.8.6",
+    "react-native-webview": "^11.23.1",
```
Comment (theutku) [inline package.json:59]: why downgrade this?

File: src/components/ShareButton/ShareButton.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+export const ShareButton: React.FC<ShareButtonProps> = ({ item }) => {
+  const onSharePress = async () => {
+    const urls: string[] = []
+
+    if (item) {
+      item?.pages.map((item) => urls.push(item.content_urls.mobile.page))
+
+      const shareOptions = {
+        title: item?.pages[0].title,
```
Comment (theutku) [inline src/components/ShareButton/ShareButton.tsx:21]: I think you should create a deeplink to the app, otherwise you will limit the app usage

Comment (theutku) [reply src/components/ShareButton/ShareButton.tsx:21]: Also please memoize these

**Comment by theutku** [comment]
Also irrelevant, but I think we should add haptic feedback for toggling favorites

File: src/components/SegmentedTabs/SegmentedTabsSwitcherItem.tsx (hunk 2 above)
Comment (sametcl2) [reply src/components/SegmentedTabs/SegmentedTabsSwitcherItem.tsx:53]: it gave an error like "only one child acceptable" so I added this but now I removed but it's still working, so I guess RN trolled me :D

Comment (theutku) [reply src/components/SegmentedTabs/SegmentedTabsSwitcherItem.tsx:53]: no it is a react feature, so understandable

**Comment by theutku** [comment]
> Also irrelevant, but I think we should add haptic feedback for toggling favorites

Opened a PR for this

File: src/components/elements/DatePicker/DatePicker.tsx (hunk 4 above)
Comment (sametcl2) [reply src/components/elements/DatePicker/DatePicker.tsx:14]: it's opening date picker automatically on a loop on android, so it should be conditional in order to fix IOS do not has this problem, when i make it conditionally either it behaves same, alert is not shown

File: package.json (hunk 5 above)
Comment (sametcl2) [reply package.json:59]: it was giving error on android

File: src/components/elements/DatePicker/DatePicker.tsx (hunk 4 above)
Comment (theutku) [reply src/components/elements/DatePicker/DatePicker.tsx:14]: still you should use the utils

**Pull request merged**